server/src/main/java/org/apache/iotdb/db/mpp/plan/analyze/PartitionCache.java

```diff
@@ -0,0 +1,757 @@
+          new HashMap<>();
+
+      // check cache for each storage group
+      for (Map.Entry<String, List<String>> entry : storageGroupToDeviceMap.entrySet()) {
+        String storageGroup = entry.getKey();
+        Map<TSeriesPartitionSlot, TRegionReplicaSet> regionReplicaSetMap =
+            schemaPartitionMap.computeIfAbsent(storageGroup, k -> new HashMap<>());
+        SchemaPartitionTable schemaPartitionTable = schemaPartitionCache.getIfPresent(storageGroup);
+        if (null == schemaPartitionTable) {
+          // if storage group not find, then return cache miss.
+          logger.debug(
+              "[{} Cache] miss when search storage group {}",
+              SCHEMA_PARTITION_CACHE_NAME,
+              storageGroup);
+          CacheMetricsRecorder.record(false, SCHEMA_PARTITION_CACHE_NAME);
+          return null;
+        }
+        Map<TSeriesPartitionSlot, TConsensusGroupId> map =
+            schemaPartitionTable.getSchemaPartitionMap();
+        // check cache for each device
+        for (String device : entry.getValue()) {
+          TSeriesPartitionSlot seriesPartitionSlot =
+              partitionExecutor.getSeriesPartitionSlot(device);
+          if (!map.containsKey(seriesPartitionSlot)) {
+            // if one device not find, then return cache miss.
+            logger.debug(
+                "[{} Cache] miss when search device {}", SCHEMA_PARTITION_CACHE_NAME, device);
+            CacheMetricsRecorder.record(false, SCHEMA_PARTITION_CACHE_NAME);
+            return null;
+          }
+          TConsensusGroupId consensusGroupId = map.get(seriesPartitionSlot);
+          TRegionReplicaSet regionReplicaSet = getRegionReplicaSet(consensusGroupId);
+          regionReplicaSetMap.put(seriesPartitionSlot, regionReplicaSet);
+        }
+      }
+      logger.debug("[{} Cache] hit", SCHEMA_PARTITION_CACHE_NAME);
+      // cache hit
+      CacheMetricsRecorder.record(true, SCHEMA_PARTITION_CACHE_NAME);
+      return new SchemaPartition(
+          schemaPartitionMap, seriesSlotExecutorName, seriesPartitionSlotNum);
+    } finally {
+      schemaPartitionCacheLock.readLock().unlock();
+    }
+  }
+
+  /** update schemaPartitionCache by schemaPartition. */
+  public void updateSchemaPartitionCache(
+      Map<String, Map<TSeriesPartitionSlot, TConsensusGroupId>> schemaPartitionTable) {
+    schemaPartitionCacheLock.writeLock().lock();
+    try {
+      for (Map.Entry<String, Map<TSeriesPartitionSlot, TConsensusGroupId>> entry1 :
+          schemaPartitionTable.entrySet()) {
+        String storageGroup = entry1.getKey();
+        SchemaPartitionTable result = schemaPartitionCache.getIfPresent(storageGroup);
+        if (null == result) {
+          result = new SchemaPartitionTable();
+          schemaPartitionCache.put(storageGroup, result);
+        }
+        Map<TSeriesPartitionSlot, TConsensusGroupId> result2 = result.getSchemaPartitionMap();
```

Comment:
parameter‘s name should be more specific

Reply:
Fixed~